### ISSUE TITLE
Custom workflow labels

### ISSUE THREAD

**Issue by al-vovk** [promhippie/github_exporter#339]
Hi there, is there any way to add custom labels to `workflow` metrics?

I was using this (https://promhippie.github.io/github_exporter/#getting-started) reference doc to set the exporter, and it mentioned the following ENV variable:

```
GITHUB_EXPORTER_WORKFLOWS_LABELS
List of labels used for workflows, comma-separated list, defaults to owner, repo, workflow, event, name, status, branch, number, run
```

Is it possible to enrich the default set of labels, or is this ENV variable there to reduce cardinality? 

I would like to tie the `runner_name` from the webhook payload to the `workflow` metrics to gain insight into our self-hosted runner's usage.

**Comment by tboerger** [comment]
Generally this variable exist to reduce cardinality, but it could also get extended.

This list of available labels is defined at https://github.com/promhippie/github_exporter/blob/master/pkg/config/config.go#L93 while the mapping is handled by https://github.com/promhippie/github_exporter/blob/master/pkg/store/types.go#L27.

If the webhook parses more flags and stores it within the database this could be extended to get mapped.

**Comment by tboerger** [comment]
If this doesn't answer your question feel free to reopen this issue, otherwise I gonna close this issue for now.

**Comment by apj0nes** [comment]
@tboerger I have the exact same use-case as described by @al-vovk.  
We are trying to identify which runners are running which workflows.  
Having the runner_name label added to the workflows would enable this.  
The docs lead me to believe that I would be able to add any field returned by the github webhook as a label, but this is not the case and needs to be added to the case statement https://github.com/promhippie/github_exporter/blob/master/pkg/store/types.go#L27

Would you be open to supporting this, happy to contribute.

**Comment by apj0nes** [comment]
Hey @tboerger 
Would you be open to a contribution on this?
Essentially supporting the WorkflowJobEvent and emitting metrics.

**Comment by tboerger** [comment]
I would be open to contributons, absolutely! If this is increasing the load or the count of labels we should just add flags to enable/disable the metrics and disable it by default :)

**Comment by nikolaik** [comment]
Hi @apj0nes did you ever start looking at handling `workflow_run` webhook events and emitting `github_workflow_job_*` type metrics? I'm interested in the same use case and I'm just about to start looking at it

**Comment by tboerger** [comment]
This should be resolved with the release of v4.0.0, if something is missing or not working please raise an issue. The release should pop up in a few minutes.
